julia/src/BridgeStan.jl

```diff
@@ -0,0 +1,336 @@
+
+        if !isfile(stanlib_)
+            throw(SystemError("Dynamic library file not found"))
+        end
+
+        if datafile_ != "" && !isfile(datafile_)
+            throw(SystemError("Data file not found"))
+        end
+
+        lib = Libc.Libdl.dlopen(stanlib_)
+
+        stanmodel = ccall(
+            Libc.Libdl.dlsym(lib, "construct"),
+            Ptr{StanModelStruct},
+            (Cstring, UInt32, UInt32),
+            datafile_,
+            seed,
+            chain_id,
+        )
+        if stanmodel == C_NULL
+            error("could not construct model RNG")
+        end
+
+        sm = new(lib, stanmodel, datafile_, seed, chain_id)
+
+        function f(sm)
+            ccall(
+                Libc.Libdl.dlsym(sm.lib, "destruct"),
+                UInt32,
+                (Ptr{StanModelStruct},),
+                sm.stanmodel,
+            )
+        end
+
+        finalizer(f, sm)
+    end
+end
+
+function name(sm::StanModel)
+    cstr = ccall(
+        Libc.Libdl.dlsym(sm.lib, "name"),
+        Cstring,
+        (Ptr{StanModelStruct},),
+        sm.stanmodel,
+    )
+    unsafe_string(cstr)
+end
+
+function param_num(sm::StanModel; include_tp = false, include_gq = false)
+    ccall(
+        Libc.Libdl.dlsym(sm.lib, "param_num"),
+        Cint,
+        (Ptr{StanModelStruct}, Cint, Cint),
+        sm.stanmodel,
+        include_tp,
+        include_gq,
+    )
+end
+
+function param_unc_num(sm::StanModel)
+    ccall(
+        Libc.Libdl.dlsym(sm.lib, "param_unc_num"),
+        Cint,
+        (Ptr{StanModelStruct},),
+        sm.stanmodel,
+    )
+end
+
+function param_names(sm::StanModel; include_tp = false, include_gq = false)
+    cstr = ccall(
+        Libc.Libdl.dlsym(sm.lib, "param_names"),
+        Cstring,
+        (Ptr{StanModelStruct}, Cint, Cint),
+        sm.stanmodel,
+        include_tp,
+        include_gq,
+    )
+    [string(s) for s in split(unsafe_string(cstr), ',')]
+end
+
+function param_unc_names(sm::StanModel)
+    cstr = ccall(
+        Libc.Libdl.dlsym(sm.lib, "param_unc_names"),
+        Cstring,
+        (Ptr{StanModelStruct},),
+        sm.stanmodel,
+    )
+    [string(s) for s in split(unsafe_string(cstr), ',')]
+end
+
+function param_constrain!(
+    sm::StanModel,
+    theta_unc,
+    out::Vector{Float64};
+    include_tp = false,
+    include_gq = false,
+)
+    dims = param_num(sm; include_tp = include_tp, include_gq = include_gq)
+    if length(out) != dims
+        throw(
+            DimensionMismatch("out must be same size as number of constrained parameters"),
+        )
+    end
+    rc = ccall(
+        Libc.Libdl.dlsym(sm.lib, "param_constrain"),
+        Cint,
+        (Ptr{StanModelStruct}, Cint, Cint, Ref{Cdouble}, Ref{Cdouble}),
+        sm.stanmodel,
+        include_tp,
+        include_gq,
+        theta_unc,
+        out,
+    )
+    if rc != 0
+        error("param_constrain failed on C++ side; see stderr for messages")
+    else
```

Comment:
Change or not, as this is definitely a personal preference.

I'd prefer to end the if and return immediately after, instead of returning from within the else block.

Reply:
Whichever you pick, will you make it consistent throughout?